### PR TITLE
Updated Diagnostics/Diagnostics_Interface.F90 to write basic grid inf…

### PR DESCRIPTION
**Pull request by illorenzo7**
…ormation to binary file 'grid_info' and added routine 'GridInfo' to post_processing/rayleigh_diagnostics.py to read it.

This will put the file 'grid_info' (containing basic information about the computational grid: collocation points, etc.) into the  root directory of each Rayleigh run during initialization (before the first time step). Variables included are the Chebyshev/legendre collocation points (r and theta), derivative quantities, and the radial and latitudinal integration weights, rweights and tweights, which are used by Rayleigh transform Spherical Averages into Global Averages and Azimuthal Averages into Spherical Averages, respectively.

### CODE REVIEW

**Comment by feathern** [comment]
Hi Loren,
I completely missed this.  I will try to have a look within the next week.
-Nick

File: post_processing/rayleigh_diagnostics.py [hunk 1]
```diff
@@ -350,6 +350,58 @@ def __init__(self,filename='none',path='./'):
...
+    self.sintheta   : sin(theta)
+    self.tweights   : latitudinal integration weights
+    self.phi        : zonal coordinates
+    self.dphi       : zonal integration weight(s) (all 2*PI/n_phi)
+    """
+
+    def __init__(self, filename='none', path='./'):
+        """filename  : The reference state file to read.
+           path      : The directory where the file is located \
+                   (if full path not in filename
```
Comment (gassmoeller) [inline post_processing/rayleigh_diagnostics.py]: Is there a closing parentheses missing here?

File: post_processing/rayleigh_diagnostics.py [hunk 2]
```diff
@@ -350,6 +350,58 @@ def __init__(self,filename='none',path='./'):
...
+    self.n_theta    : number of latitudinal points
+    self.n_phi      : number of zonal points
+    self.radius     : radial coordinates
+    self.rweights   : radial integration weights
+    self.theta      : (co)latitudinal coordinates
+    self.costheta   : cos(theta)
+    self.sintheta   : sin(theta)
+    self.tweights   : latitudinal integration weights
+    self.phi        : zonal coordinates
+    self.dphi       : zonal integration weight(s) (all 2*PI/n_phi)
```
Comment (gassmoeller) [inline post_processing/rayleigh_diagnostics.py]: Can this be of length 1 or n_phi? Could you make that clear here?

File: post_processing/rayleigh_diagnostics.py [hunk 3]
```diff
@@ -350,6 +350,58 @@ def __init__(self,filename='none',path='./'):
...
+                swap=bs)
+        self.phi = swapread(fd, dtype='float64', count=nphi,\
+                swap=bs)
+        self.dphi = swapread(fd, dtype='float64', count=1, swap=bs)
+        self.names = ['nr', 'ntheta', 'nphi', 'radius', 'rweights',\
+                'theta', 'costheta', 'sintheta', 'tweights', 'phi',\
+                'dphi']        
+
+        fd.close()
 class G_Avgs:
```
Comment (gassmoeller) [inline post_processing/rayleigh_diagnostics.py]: Please leave an empty line here before the next class.

File: src/Diagnostics/Diagnostics_Interface.F90 [hunk 4]
```diff
@@ -271,8 +271,9 @@ End Subroutine Read_Output_Namelist
 
     Subroutine Initialize_Diagnostics()
         Implicit None
-        Integer :: i, isize
+        Integer :: i, isize, n_phi, sig=314
```
Comment (gassmoeller) [inline src/Diagnostics/Diagnostics_Interface.F90]: Maybe this is my lack of understanding of fortran, but what is `sig=314`? Is there a more descriptive name? Or can you add a comment that explains what this variable is, and why it is 314?

**Pull request closed without merging**